### PR TITLE
Implement `.log` to log map property changes

### PR DESCRIPTION
**Pull request by m-mujica**
Example output:

screen shot 2017-10-17 at 15 05 12

### CODE REVIEW

**Comment by justinbmeyer** [comment]
Can you line up the values?

**Comment by justinbmeyer** [comment]
Add a space after ‘is’

**Comment by m-mujica** [comment]
Sure thing,

screen shot 2017-10-17 at 16 41 54

looking pretty now.

**Comment by m-mujica** [comment]
@justinbmeyer I changed this so it does not rely on the changes of `can-event-queue`.

There is only two places where `dispatch` is called so I figured it was ok to just called `this._log` next to the propertyName dispatch call.

QUESTION: do I need a `Set` polyfill or is it ok to just do: `new Set` ?

File: can-simple-map.js [hunk 1]
```diff
@@ -84,8 +103,37 @@ var SimpleMap = Construct.extend("SimpleMap",
...
+		// call `.log()` to log all property changes
+		// pass a single property to only get logs for said property, e.g: `.log("foo")`
+		log: function(key) {
+			//!steal-remove-start
+			var quoteString = function quoteString(x) {
+				return typeof x === "string" ? JSON.stringify(x) : x;
+			};
+
+			var meta = ensureMeta(this);
+			meta.allowedLogKeysSet = meta.allowedLogKeysSet || new Set();
```
Comment (m-mujica) [inline can-simple-map.js]: Do I need to polyfill `Set` for older browsers?

**Pull request merged**